Documentation/learning/glossary.md

```diff
@@ -2,22 +2,70 @@
 
 This document defines the various terms used in etcd documentation, command line and source code.
 
-## Node
+## Alarm
 
-Node is an instance of raft state machine.
+Alarms are a means to recognize any fault in the etcd server.
 
-It has a unique identification, and records other nodes' progress internally when it is the leader.
+## Authentication
 
-## Member
+Authentication is used to authenticate a user before the user can perform a set of operations.
 
-Member is an instance of etcd. It hosts a node, and provides service to clients.
+## Client
+
+Client is a caller of the cluster's HTTP API.
 
 ## Cluster
 
 Cluster consists of several members.
 
 The node in each member follows raft consensus protocol to replicate logs. Cluster receives proposals from members, commits them and apply to local store.
 
+## Command
+
+Command is an identifiers which are used to perform a specific operation in etcd server using etcdctl.
+Example: put, get, del, lease, etc.
+
+## Compaction
+
+Compaction discards all etcd event history prior to a given revision. Since etcd uses a multiversion concurrency control model, it preserves all key updates as event history. When the event history up to some revision is no longer needed, all superseded keys may be compacted away to reclaim storage space in the etcd backend database.
+
+## Election
+
+Election is an in-build functionality of etcd-server which facilitates in electing a leader in etcd server cluster.
+
+## Endpoint
+
+Endpoint refers to the target member where the client request or operation needs to be done.
+
+## Keys
+
+Keys are unique idetifiers. Keys are used to store values to the store and the values can be retrieved from the store using the keys.
+
+## Keyspaces
+
+Keyspaces refer to the set of all possible permutations of a key.
+
+## Lease
+
+Lease is a contract given for a specified duration. It can be used to attach to the keys to make keys available as long as the lease lasts. Once the lease expires all the keys associated with it also expire.
+
+## Member
+
+Member is an instance of etcd. It hosts a node, and provides service to clients.
+
+## Mirroring
+
+Mirroring is a process of mirroring a prefix in the key-value space of an etcd cluster into another prefix in another cluster. Mirroring is designed for copying configuration to various clusters distributed around the world.
+
+## ModRevision
+
+ModRevision is the etcd revision of the last update done on a key. Every time a key gets updated the modrevision gets incremented by 1.
```

Comment:
@heyitsanthony the definition you provided may be confusing to the new users, can you simplify it so that its more simple to understand for new users?